Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -295,16 +436,30 @@ public struct HTTPHeaders: CustomStringConvertible {
     /// - Parameter name: The header field name whose values are to be retrieved.
     /// - Returns: A list of the values for that header field name.
     public func getCanonicalForm(_ name: String) -> [String] {
+        let result = self[name]
+
         // It's not safe to split Set-Cookie on comma.
-        let queryName = name.lowercased()
-        if queryName == "set-cookie" {
-            return self[name]
+        guard name.lowercased() != "set-cookie" else {
+            return result
         }
 
-        if let result = storage[queryName] {
-            return result.map { tuple in tuple.1.split(separator: ",").map { String($0.trimWhitespace()) } }.reduce([], +)
+        return result.map { $0.split(separator: ",").map { String($0.trimWhitespace()) } }.reduce([], +)
+    }
+}
+
+private enum HeaderName {
+    case string(string: String)
+    case staticString(string: StaticString)
+
+    func validateAndWrite(buffer: inout ByteBuffer) -> Int {
+        switch self {
+        case .staticString(let string):
+            precondition(string.isASCII, "name must be ASCII")
```

Comment:
Where is `StaticString.isASCII` coming from? Why can't we use it on `string` below?

Reply:
https://developer.apple.com/documentation/swift/staticstring/1539231-isascii its part of it ;)